htlcswitch/link.go

```diff
@@ -2144,6 +2157,27 @@ func (l *channelLink) handleUpstreamMsg(msg lnwire.Message) {
 			"ChannelPoint(%v): received error from peer: %v",
 			l.channel.ChannelPoint(), msg.Error(),
 		)
+
+	case *lnwire.Shutdown:
+		// We've received a Shutdown message from the remote peer.
+		// We'll set the shutdownReceived bool and cancel back any new
+		// HTLC's received after this point instead of forwarding them.
+		if l.shutdownReceived.Load() {
+			// Fail the link if a duplicate shutdown is received.
+			// This is also checked in peer.Brontide, but done here
+			// to properly stop the link.
+			l.fail(
+				LinkFailureError{code: ErrRemoteError},
+				"ChannelPoint(%v): received dupe shutdown "+
+					"from peer", l.channel.ChannelPoint(),
+			)
+
+			return
+		}
+
+		// Otherwise, set the shutdownReceived bool.
+		l.shutdownReceived.Store(true)
```

Comment:
Sent but then it doesn't cache the message? See the comment above. I think generally the link doesn't need to know about these messages. Only that "something" wants to shut down, which we already have the pipeline for.